python/src/aiconfig/editor/client/src/components/aiconfigReducer.ts

```diff
@@ -50,11 +67,60 @@ function reduceReplaceInput(
   }));
 }
 
+function reduceInsertPromptAtIndex(
+  state: ClientAIConfig,
+  index: number,
+  prompt: ClientPrompt
+): ClientAIConfig {
+  return {
+    ...state,
+    prompts: [
+      ...state.prompts.slice(0, index),
+      prompt,
+      ...state.prompts.slice(index),
+    ],
+  };
+}
+
+function reduceConsolidateAIConfig(
+  state: ClientAIConfig,
+  action: MutateAIConfigAction,
+  responseConfig: AIConfig
+): ClientAIConfig {
+  switch (action.type) {
+    case "ADD_PROMPT_AT_INDEX": {
```

Comment:
I don't necessarily think any reconsolidation is necessary after adding a prompt, but wanted to have this as an example. Some actions could potentially update the config in the server and we should make sure those changes are subsequently reflected on the client as well. We should still prioritize client-side input/metadata since it can be changed on the UI while the server requests are still in flight, so we don't want to overwrite the client with the server state for those values.

As a side note, I realize this is going to be problematic as-is since add_prompt on the config size doesn't support specifying an index yet. So, we can add at any index on the client but only the end on the server. I'll update the code to just have the add prompt in the UI at the bottom for now

Reply:
Done